app/uk/gov/hmrc/economiccrimelevyaccount/connectors/ObligationDataConnector.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyaccount.connectors
+
+import play.api.Logging
+import uk.gov.hmrc.economiccrimelevyaccount.config.AppConfig
+import uk.gov.hmrc.economiccrimelevyaccount.models.ObligationData
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ObligationDataConnector @Inject() (appConfig: AppConfig, httpClient: HttpClient)(implicit ec: ExecutionContext)
+    extends Logging {
+
+  private val eclAccountUrl: String = s"${appConfig.economicCrimeLevyAccountBaseUrl}/economic-crime-levy-account"
+
+  def getObligationData()(implicit hc: HeaderCarrier): Future[Option[ObligationData]] =
+    httpClient.GET[Option[ObligationData]](
+      s"$eclAccountUrl/obligation-data"
```

Comment:
How does the API know which ECL account we are requesting the obligation data for? Normally a GET will have the identifier in the Url, won't it?

Reply:
I've just utilised same request that is already being used before and is authorised for that base context path (`GET /economic-crime-levy-account`). Take look at [this call](https://github.com/hmrc/economic-crime-levy-account-frontend/pull/6/files#diff-1a2fa0bcc513b94bbd15653cebcd6035297357d9574d4577fece63ef2f541780R44). I'm reusing same request (2 lines before) because it's for same ecl reference.

Reply:
If you take a look [here](https://github.com/hmrc/economic-crime-levy-account/blob/main/app/uk/gov/hmrc/economiccrimelevyaccount/controllers/ObligationDataController.scala#L36) BE is pulling eclReference from request (that I have just recycled) and making call further.

Reply:
Yes, of course :)